### PR TITLE
Adding a file

### PR DESCRIPTION
**Pull request by antuanettaV**
Please kindly check my homework (week4). Thanks in advance.

### CODE REVIEW

File: text.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+</head>
+<body>
+<script src="week.js"></script>
```
Comment (vdaguenet) [inline text.html:9]: This file is not the one where you implemented week 4's homework.

File: java.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    state.name = name;
+
+    return `Nice to meet you, ${name}.`;
+  } else if (command === "What is my name?") {
+    if (state.name) {
+      return `Your name is ${state.name}.`;
+    } else {
+      return "Sorry, I do not know your name.";
+    }
+  } else if (command === "Add fishing to my todo") {
```
Comment (vdaguenet) [inline java.js:18]: `Add fishing to my todo` and `Add singing in the shower to my todo` should be handled in the same case and in a way that would work with `Add whatever to my todo`, too.

**Review by vdaguenet — CHANGES_REQUESTED** [review]
Unfortunately, the to-do list section of the homework was misunderstood. Please make the necessary adjustments, and I will have a second look at your solution.

**Comment by antuanettaV** [comment]
Good day, Valentin. Thank you for your feedback. I have changed my homework. Could you please kindly have a look once you will have time? Thanks in advance.

File: java.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+let state = {
```
Comment (vdaguenet) [inline java.js:1]: Nice idea!

File: java.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  name: "",
+  todoList: [],
+};
+
+function getReply(command) {
+  if (command.startsWith("Hello my name is ")) {
```
Comment (vdaguenet) [inline java.js:7]: This works fine in the context of the exercise. When dealing with real user inputs, it's recommended to clean up the command like this:
```js
const cleanCommand = command.toLowerCase().trim();
if (cleanCommand.startsWith('hello my name is')) { ... }
```

Doing so allows your code to run even if the user made a typo like `Hello my Name Is Valentin`

File: java.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    if (state.todoList.length === 0) {
+      return "There is nothing in your todo.";
+    } else {
+      const todoItems = state.todoList.join(" and ");
+      return `You have ${state.todoList.length} activity(ies) - ${todoItems}.`;
+    }
+  } else if (command === "What day is it today?") {
+    const today = new Date();
+    const options = { day: "numeric", month: "long", year: "numeric" };
+    const formattedDate = today.toLocaleDateString("en-US", options);
```
Comment (vdaguenet) [inline java.js:46]: Nice way to format a date.

**Review by vdaguenet — APPROVED** [review]
Good work!

**Pull request merged**